contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,85 @@
+# VueJs
+
+Author: Lu Lechuan
+
+A JavaScript framework is an application framework written in JavaScript. It differs from a JavaScript library in its control flow: a library offers functions to be called by its parent code, whereas a framework defines the entire application design.
+JavaScript frameworks aim to improve code quality and maintainablity, example of some javascript frameworks include Angular, React and Vue.
+
+## Advantages of VueJs
+VueJs is an approachable, versetile and performant framework that helps to create a more maintainable and testable database.
+
+* approachable: VueJs is very easy to learn. Comparing to other framework such as Augular, Vue is simple in term of API and design. Learning enough to build non-trivial applications typically takes less than a day.
+
+* progressive: Vue is designed from the ground up to be incrementally adoptable. The core library is focused on the view layer only, and is easy to pick up and integrate with other libraries or existing projects. This means that if you hava a large application, you can plug Vue into just a part of your application without messing the rest up.
+
+* versetile: Vue is perfectly capable of powering sophisticated Single-Page Applications when used in combination with modern tooling and supporting libraries.
+
+* clean: Vue symtax is simple and this can make the HTML pages very clean.
+
+## Installation
+
+There are many ways to install VueJs.
+* Direct `<script>` Include:
+  - [development version](https://vuejs.org/js/vue.js)
+  - [product version](https://vuejs.org/js/vue.min.js)
+* CDN:
+  - [cdn](<script src="https://cdn.jsdelivr.net/npm/vue@2.5.13/dist/vue.js"></script>)
+* NPM:
+  - $ npm install vue
+* CLI:
+```cli
+    # install vue-cli
+    $ npm install --global vue-cli
+    # create a new project using the "webpack" template
+    $ vue init webpack my-project
+    # install dependencies and go!
+    $ cd my-project
+    $ npm install
+    $ npm run dev
+```
+
+Apart from installing VueJs, you can install VueJs development tools in your browser. This will allow you to inspect and debug the Vue components in your projects. The link is as follow:
+- [Vue DevTools](https://github.com/vuejs/vue-devtools#vue-devtools)
+
+## The first usage of VueJs:
+
+We first starts our need to display message on our webpage:
+
+```HTML
+<body>
+  <div id="root">
+
+		<h1>{{ message }}</h1>
+
+	</div>
+
+	<script src="https://unpkg.com/vue@2.5.13/dist/vue.js"></script>
+
+	<script src="the_path_to_the_javacript_file.js"></script>
+</body>
+```
+
+```js
+new Vue ({
+	el: '#root',
+
+  data: {
+    message: Hello World
+  }
+});
+```
+
+Explanation:
```

Comment:
- `JavaScript file which includes* our Vue codes`
- `define the message` rather than `define a message`
- `Vue` not `vue`?
- use `root` rather than `"root"` for consistency

Could you clarify what you mean by `only this component is working in Vue`? What does 'working' mean? Consider merging the point beginning with `Remember...` with this point in order to provide a more comprehensive idea of what is bound to what and how?